{"content":"\r\n  \r\n  <h1>\r\n    <span class=\"lang-csharp\">CommunicationException</span>\r\n    \t\t\t\tClass\r\n  </h1>\r\n  \r\n  <h2 class=\"hiddenAnchor\">Definition</h2>\r\n  \r\n  <div class=\"metadata\">\r\n  \r\n  \r\n  </div>\r\n  \r\n  <div class=\"summaryHolder\">\r\n  \t<div class=\"summary\">\r\n  \t\t<p>Represents a communication error in either the service or client application.</p>\n\r\n  \t</div>\r\n  </div><nav id=\"center-doc-outline\" class=\"doc-outline\" ms.cmpgrp=\"intopic toc\" role=\"navigation\" aria-label=\"On page navigation\">\r\n    <h3>In this Article</h3>\r\n  </nav>\t<pre class=\"wrap loading\"><code class=\"csharp lang-csharp\">public class CommunicationException : SystemException</code></pre>\r\n  \r\n  \r\n  \r\n  \r\n  \t<dl class=\"attributeList stack\">\r\n  \t\t<dt>Inheritance</dt><dd>\r\n  \t\t<div class=\"pdlHolder\">\r\n  \t\t<div class=\"pdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.object\" data-linktype=\"relative-path\">Object</a></span></div><div class=\"pdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception\" data-linktype=\"relative-path\">Exception</a></span></div><div class=\"pdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.systemexception\" data-linktype=\"relative-path\">SystemException</a></span></div><span class=\"lang-csharp\">CommunicationException</span>\r\n  \t\t</div>\r\n  \t\t</dd>\r\n  \t</dl>\r\n  \r\n  \r\n  \r\n  \r\n  \t<h3 class=\"accented\">Inherited Members</h3>\r\n  \t<h4 class=\"accented\"></h4>\r\n  \t<div class=\"cdlHolder\">\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception#System_Exception_Data\" data-linktype=\"relative-path\">Data</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception#System_Exception_GetBaseException\" data-linktype=\"relative-path\">GetBaseException()</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception#System_Exception_GetObjectData_System_Runtime_Serialization_SerializationInfo_System_Runtime_Serialization_StreamingContext_\" data-linktype=\"relative-path\">GetObjectData(SerializationInfo, StreamingContext)</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception#System_Exception_GetType\" data-linktype=\"relative-path\">GetType()</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception#System_Exception_HelpLink\" data-linktype=\"relative-path\">HelpLink</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception#System_Exception_HResult\" data-linktype=\"relative-path\">HResult</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception#System_Exception_InnerException\" data-linktype=\"relative-path\">InnerException</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception#System_Exception_Message\" data-linktype=\"relative-path\">Message</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception#System_Exception_SerializeObjectState\" data-linktype=\"relative-path\">SerializeObjectState</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception#System_Exception_Source\" data-linktype=\"relative-path\">Source</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception#System_Exception_StackTrace\" data-linktype=\"relative-path\">StackTrace</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception#System_Exception_TargetSite\" data-linktype=\"relative-path\">TargetSite</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception#System_Exception_ToString\" data-linktype=\"relative-path\">ToString()</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.object#System_Object_Equals_System_Object_\" data-linktype=\"relative-path\">Equals(Object)</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.object#System_Object_Equals_System_Object_System_Object_\" data-linktype=\"relative-path\">Equals(Object, Object)</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.object#System_Object_GetHashCode\" data-linktype=\"relative-path\">GetHashCode()</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.object#System_Object_MemberwiseClone\" data-linktype=\"relative-path\">MemberwiseClone()</a></span></div>\r\n  \t\t\t<div class=\"cdl\"><span class=\"lang-csharp\"><a class=\"xref\" href=\"system.object#System_Object_ReferenceEquals_System_Object_System_Object_\" data-linktype=\"relative-path\">ReferenceEquals(Object, Object)</a></span></div>\r\n  \t</div>\r\n  \r\n  \t<h2 class=\"accented\">Examples</h2>\r\n  \t<p>The following code example shows a client that handles CommunicationException types. This client also handles <a class=\"xref\" href=\"system.servicemodel.faultexception\" data-linktype=\"relative-path\">FaultException</a> objects because the service has <a class=\"xref\" href=\"system.servicemodel.servicebehaviorattribute#System_ServiceModel_ServiceBehaviorAttribute_IncludeExceptionDetailInFaults_\" data-linktype=\"relative-path\">IncludeExceptionDetailInFaults</a> set to <code>true</code>.  </p>\n<pre><code class=\"lang-cs\" name=\"IErrorHandler#3\">using System;\nusing System.ServiceModel;\nusing System.ServiceModel.Channels;\nusing Microsoft.WCF.Documentation;\n\npublic class Client\n{\n  public static void Main()\n  {\n    // Picks up configuration from the configuration file.\n    SampleServiceClient wcfClient = new SampleServiceClient();\n    try\n    {\n      // Making calls.\n      Console.WriteLine(&quot;Enter the greeting to send: &quot;);\n      string greeting = Console.ReadLine();\n      Console.WriteLine(&quot;The service responded: &quot; + wcfClient.SampleMethod(greeting));\n      Console.WriteLine(&quot;Press ENTER to exit:&quot;);\n      Console.ReadLine();\n    }\n    catch (TimeoutException timeProblem)\n    {\n      Console.WriteLine(&quot;The service operation timed out. &quot; + timeProblem.Message);\n      wcfClient.Abort();\n      Console.ReadLine();\n    }\n    // Catch the contractually specified SOAP fault raised here as an exception.\n    catch (FaultException&lt;GreetingFault&gt; greetingFault)\n    {\n      Console.WriteLine(greetingFault.Detail.Message);\n      Console.Read();\n      wcfClient.Abort();\n    }\n    // Catch unrecognized faults. This handler receives exceptions thrown by WCF\n    // services when ServiceDebugBehavior.IncludeExceptionDetailInFaults \n    // is set to true.\n    catch (FaultException faultEx)\n    {\n      Console.WriteLine(&quot;An unknown exception was received. &quot; \n        + faultEx.Message\n        + faultEx.StackTrace\n      );\n      Console.Read();\n      wcfClient.Abort();\n    }\n    // Standard communication fault handler.\n    catch (CommunicationException commProblem)\n    {\n      Console.WriteLine(&quot;There was a communication problem. &quot; + commProblem.Message + commProblem.StackTrace);\n      Console.Read();\n      wcfClient.Abort();\n    }\n  }\n}\n</code></pre><pre><code class=\"lang-vb\" name=\"IErrorHandler#3\">\nImports System\nImports System.ServiceModel\nImports System.ServiceModel.Channels\nImports Microsoft.WCF.Documentation\n\nPublic Class Client\n  Public Shared Sub Main()\n\t&#39; Picks up configuration from the configuration file.\n\tDim wcfClient As New SampleServiceClient()\n\tTry\n\t  &#39; Making calls.\n\t  Console.WriteLine(&quot;Enter the greeting to send: &quot;)\n\t  Dim greeting As String = Console.ReadLine()\n\t  Console.WriteLine(&quot;The service responded: &quot; &amp; wcfClient.SampleMethod(greeting))\n\t  Console.WriteLine(&quot;Press ENTER to exit:&quot;)\n\t  Console.ReadLine()\n\tCatch timeProblem As TimeoutException\n\t  Console.WriteLine(&quot;The service operation timed out. &quot; &amp; timeProblem.Message)\n\t  wcfClient.Abort()\n\t  Console.ReadLine()\n\t&#39; Catch the contractually specified SOAP fault raised here as an exception.\n\tCatch greetingFault As FaultException(Of GreetingFault)\n\t  Console.WriteLine(greetingFault.Detail.Message)\n\t  Console.Read()\n\t  wcfClient.Abort()\n\t&#39; Catch unrecognized faults. This handler receives exceptions thrown by WCF\n\t&#39; services when ServiceDebugBehavior.IncludeExceptionDetailInFaults \n\t&#39; is set to true.\n\tCatch faultEx As FaultException\n\t  Console.WriteLine(&quot;An unknown exception was received. &quot; &amp; faultEx.Message + faultEx.StackTrace)\n\t  Console.Read()\n\t  wcfClient.Abort()\n\t&#39; Standard communication fault handler.\n\tCatch commProblem As CommunicationException\n\t  Console.WriteLine(&quot;There was a communication problem. &quot; &amp; commProblem.Message + commProblem.StackTrace)\n\t  Console.Read()\n\t  wcfClient.Abort()\n\tEnd Try\n  End Sub\nEnd Class\n</code></pre>\r\n  \r\n  \t<h2 class=\"accented\">Remarks</h2>\r\n  \t<p>Robust client and service Windows Communication Foundation (WCF) applications handle CommunicationException objects that may be thrown during communication. There are also two CommunicationException-derived exception types (<a class=\"xref\" href=\"system.servicemodel.faultexception-1\" data-linktype=\"relative-path\">FaultException&lt;TDetail&gt;</a> and <a class=\"xref\" href=\"system.servicemodel.faultexception\" data-linktype=\"relative-path\">FaultException</a>) that clients also often expect. Therefore, in order to prevent the generic CommunicationException handler from catching these more specific exception types, catch these exceptions prior to handling CommunicationException.  </p>\n<ul>\n<li><a class=\"xref\" href=\"system.servicemodel.faultexception-1\" data-linktype=\"relative-path\">FaultException&lt;TDetail&gt;</a> objects are thrown on the client when a SOAP fault that is specified in the operation contract is received in response to a two-way operation (that is, a method with an <a class=\"xref\" href=\"system.servicemodel.operationcontractattribute\" data-linktype=\"relative-path\">OperationContractAttribute</a> attribute with <a class=\"xref\" href=\"system.servicemodel.operationcontractattribute#System_ServiceModel_OperationContractAttribute_IsOneWay_\" data-linktype=\"relative-path\">IsOneWay</a> set to <code>false</code>).  </li>\n</ul>\n<p> <a class=\"xref\" href=\"system.servicemodel.faultexception\" data-linktype=\"relative-path\">FaultException</a> objects are thrown when a listener receives a SOAP fault that is not expected or specified in the operation contract. This usually occurs when the application is being debugged and the service has the <a class=\"xref\" href=\"system.servicemodel.servicebehaviorattribute#System_ServiceModel_ServiceBehaviorAttribute_IncludeExceptionDetailInFaults_\" data-linktype=\"relative-path\">IncludeExceptionDetailInFaults</a> property set to <code>true</code>.  </p>\n<div class=\"NOTE\"><h5>Note</h5><p> When implementing custom channels and binding elements, it is strongly recommended that your components throw only <a class=\"xref\" href=\"system.timeoutexception\" data-linktype=\"relative-path\">System.TimeoutException</a> or CommunicationException-derived objects. In the case where your components throw a recoverable exception that is specific to the component, wrap that exception inside a CommunicationExceptionobject.  </p>\n</div>\n<p> For more details about designing and using the WCF fault system, see <a href=\"../topic/specifying%20and%20handling%20faults%20in%20contracts%20and%20services\" data-linktype=\"relative-path\">Specifying and Handling Faults in Contracts and Services</a>.  </p>\n<div class=\"IMPORTANT\"><h5>Important</h5><p> The WCF Runtime will not throw a CommunicationException that is unsafe to handle at the point where it leaves the WCF Runtime and enters user code.</p>\n</div>\n\r\n  \r\n  \r\n  \t\t<section>\r\n  \t\t\t<h2 class=\"accented \">Constructors\r\n  \t\t</h2>\r\n  \t\t\r\n  \t\t\t\t\t<div data-moniker=\" netframework-3.0 netframework-3.5-client netframework-3.5 netframework-4.0-client netframework-4.0 netframework-4.5.1 netframework-4.5.2 netframework-4.5 netframework-4.6.1 netframework-4.6.2 netframework-4.6\">\r\n  \t\t\r\n  \t\t\t\t<div class=\"memberNameHolder\" id=\"System_ServiceModel_CommunicationException__ctor\">\r\n  \t\t\t\t\t<h3 class=\"accented\">\r\n  \t\t\t\t\t\t<span class=\"lang-csharp\">CommunicationException()</span>\r\n  \t\t\t\t\t</h3>\r\n  \t\t\t\t</div>\r\n  \t\t\r\n  \t\t\t\t<div class=\"memberInfo\">\r\n  \t\t\t\t\t<p>Initializes a new instance of the <a class=\"xref\" href=\"system.servicemodel.communicationexception\" data-linktype=\"relative-path\">CommunicationException</a> class.</p>\n\r\n  \t\t\r\n  \t\t\t\t\t\t<pre class=\"wrap loading\"><code class=\"csharp lang-csharp\" dir=\"ltr\">public CommunicationException ();</code></pre>\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\t\t</div>\r\n  \t\t\t</div>\r\n  \t\t\t\t\t<div data-moniker=\" netframework-3.0 netframework-3.5-client netframework-3.5 netframework-4.0-client netframework-4.0 netframework-4.5.1 netframework-4.5.2 netframework-4.5 netframework-4.6.1 netframework-4.6.2 netframework-4.6\">\r\n  \t\t\r\n  \t\t\t\t<div class=\"memberNameHolder\" id=\"System_ServiceModel_CommunicationException__ctor_System_String_\">\r\n  \t\t\t\t\t<h3 class=\"accented\">\r\n  \t\t\t\t\t\t<span class=\"lang-csharp\">CommunicationException(String)</span>\r\n  \t\t\t\t\t</h3>\r\n  \t\t\t\t</div>\r\n  \t\t\r\n  \t\t\t\t<div class=\"memberInfo\">\r\n  \t\t\t\t\t<p>Initializes a new instance of the <a class=\"xref\" href=\"system.servicemodel.communicationexception\" data-linktype=\"relative-path\">CommunicationException</a> class, using the specified message.</p>\n\r\n  \t\t\r\n  \t\t\t\t\t\t<pre class=\"wrap loading\"><code class=\"csharp lang-csharp\" dir=\"ltr\">public CommunicationException (string message);</code></pre>\r\n  \t\t\r\n  \t\t\r\n  \t\t\t\t\t\t\t<div class=\"propertyInfoTitle stack\">Parameters</div>\r\n  \t\t\t\t\t\t\t<div class=\"propertyInfo\">\r\n  \t\t\t\t\t\t\t<dl class=\"parameterList\">\r\n  \t\t\t\t\t\t\t\t<dt>message</dt>\r\n  \t\t\t\t\t\t\t\t<dd>\r\n  \t\t\t\t\t\t\t\t\t\t<span class=\"lang-csharp\"><a class=\"xref\" href=\"system.string\" data-linktype=\"relative-path\">String</a></span>\r\n  \t\t\t\t\t\t\t\t</dd>\r\n  \t\t\t\t\t\t\t</dl>\r\n  \t\t\t\t\t\t\t<p>The description of the error condition.</p>\n\r\n  \t\t\t\t\t\t\t</div>\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\t\t</div>\r\n  \t\t\t</div>\r\n  \t\t\t\t\t<div data-moniker=\" netframework-3.0 netframework-3.5-client netframework-3.5 netframework-4.0-client netframework-4.0 netframework-4.5.1 netframework-4.5.2 netframework-4.5 netframework-4.6.1 netframework-4.6.2 netframework-4.6\">\r\n  \t\t\r\n  \t\t\t\t<div class=\"memberNameHolder\" id=\"System_ServiceModel_CommunicationException__ctor_System_Runtime_Serialization_SerializationInfo_System_Runtime_Serialization_StreamingContext_\">\r\n  \t\t\t\t\t<h3 class=\"accented\">\r\n  \t\t\t\t\t\t<span class=\"lang-csharp\">CommunicationException(SerializationInfo, StreamingContext)</span>\r\n  \t\t\t\t\t</h3>\r\n  \t\t\t\t</div>\r\n  \t\t\r\n  \t\t\t\t<div class=\"memberInfo\">\r\n  \t\t\t\t\t<p>Initializes a new instance of the <a class=\"xref\" href=\"system.servicemodel.communicationexception\" data-linktype=\"relative-path\">CommunicationException</a> class, using the specified serialization information and context objects.</p>\n\r\n  \t\t\r\n  \t\t\t\t\t\t<pre class=\"wrap loading\"><code class=\"csharp lang-csharp\" dir=\"ltr\">protected CommunicationException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);</code></pre>\r\n  \t\t\r\n  \t\t\r\n  \t\t\t\t\t\t\t<div class=\"propertyInfoTitle stack\">Parameters</div>\r\n  \t\t\t\t\t\t\t<div class=\"propertyInfo\">\r\n  \t\t\t\t\t\t\t<dl class=\"parameterList\">\r\n  \t\t\t\t\t\t\t\t<dt>info</dt>\r\n  \t\t\t\t\t\t\t\t<dd>\r\n  \t\t\t\t\t\t\t\t\t\t<span class=\"lang-csharp\"><a class=\"xref\" href=\"system.runtime.serialization.serializationinfo\" data-linktype=\"relative-path\">SerializationInfo</a></span>\r\n  \t\t\t\t\t\t\t\t</dd>\r\n  \t\t\t\t\t\t\t</dl>\r\n  \t\t\t\t\t\t\t<p>Information relevant to the deserialization process.</p>\n\r\n  \t\t\t\t\t\t\t</div>\r\n  \t\t\t\t\t\t\t<div class=\"propertyInfo stack\">\r\n  \t\t\t\t\t\t\t<dl class=\"parameterList\">\r\n  \t\t\t\t\t\t\t\t<dt>context</dt>\r\n  \t\t\t\t\t\t\t\t<dd>\r\n  \t\t\t\t\t\t\t\t\t\t<span class=\"lang-csharp\"><a class=\"xref\" href=\"system.runtime.serialization.streamingcontext\" data-linktype=\"relative-path\">StreamingContext</a></span>\r\n  \t\t\t\t\t\t\t\t</dd>\r\n  \t\t\t\t\t\t\t</dl>\r\n  \t\t\t\t\t\t\t<p>The context of the deserialization process.</p>\n\r\n  \t\t\t\t\t\t\t</div>\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\t\t</div>\r\n  \t\t\t</div>\r\n  \t\t\t\t\t<div data-moniker=\" netframework-3.0 netframework-3.5-client netframework-3.5 netframework-4.0-client netframework-4.0 netframework-4.5.1 netframework-4.5.2 netframework-4.5 netframework-4.6.1 netframework-4.6.2 netframework-4.6\">\r\n  \t\t\r\n  \t\t\t\t<div class=\"memberNameHolder\" id=\"System_ServiceModel_CommunicationException__ctor_System_String_System_Exception_\">\r\n  \t\t\t\t\t<h3 class=\"accented\">\r\n  \t\t\t\t\t\t<span class=\"lang-csharp\">CommunicationException(String, Exception)</span>\r\n  \t\t\t\t\t</h3>\r\n  \t\t\t\t</div>\r\n  \t\t\r\n  \t\t\t\t<div class=\"memberInfo\">\r\n  \t\t\t\t\t<p>Initializes a new instance of the <a class=\"xref\" href=\"system.servicemodel.communicationexception\" data-linktype=\"relative-path\">CommunicationException</a> class, using the specified message and the inner exception.</p>\n\r\n  \t\t\r\n  \t\t\t\t\t\t<pre class=\"wrap loading\"><code class=\"csharp lang-csharp\" dir=\"ltr\">public CommunicationException (string message, Exception innerException);</code></pre>\r\n  \t\t\r\n  \t\t\r\n  \t\t\t\t\t\t\t<div class=\"propertyInfoTitle stack\">Parameters</div>\r\n  \t\t\t\t\t\t\t<div class=\"propertyInfo\">\r\n  \t\t\t\t\t\t\t<dl class=\"parameterList\">\r\n  \t\t\t\t\t\t\t\t<dt>message</dt>\r\n  \t\t\t\t\t\t\t\t<dd>\r\n  \t\t\t\t\t\t\t\t\t\t<span class=\"lang-csharp\"><a class=\"xref\" href=\"system.string\" data-linktype=\"relative-path\">String</a></span>\r\n  \t\t\t\t\t\t\t\t</dd>\r\n  \t\t\t\t\t\t\t</dl>\r\n  \t\t\t\t\t\t\t<p>The description of the error condition.</p>\n\r\n  \t\t\t\t\t\t\t</div>\r\n  \t\t\t\t\t\t\t<div class=\"propertyInfo stack\">\r\n  \t\t\t\t\t\t\t<dl class=\"parameterList\">\r\n  \t\t\t\t\t\t\t\t<dt>innerException</dt>\r\n  \t\t\t\t\t\t\t\t<dd>\r\n  \t\t\t\t\t\t\t\t\t\t<span class=\"lang-csharp\"><a class=\"xref\" href=\"system.exception\" data-linktype=\"relative-path\">Exception</a></span>\r\n  \t\t\t\t\t\t\t\t</dd>\r\n  \t\t\t\t\t\t\t</dl>\r\n  \t\t\t\t\t\t\t<p>The inner exception to be used.</p>\n\r\n  \t\t\t\t\t\t\t</div>\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\r\n  \t\t\t\t</div>\r\n  \t\t\t</div>\r\n  \t\t\r\n  \t\t</section>\r\n  \r\n  \r\n","outputRootRelativePath":"../","pageMetadata":"<meta name=\"author\" content=\"dotnet-bot\">\r\n<meta name=\"ms.author\" content=\"dotnetcontent\">\r\n<meta name=\"manager\" content=\"wpickett\">\r\n<meta name=\"breadcrumb_path\" content=\"../toc2.json\">\r\n<meta name=\"ms.topic\" content=\"managed-reference\">\r\n<meta name=\"ms.prod\" content=\".net\">\r\n<meta name=\"search.ms_sitename\" content=\"Docs\">\r\n<meta name=\"search.ms_docsetname\" content=\"docs-internal\">\r\n<meta name=\"version\" content=\"0\">\r\n<meta name=\"locale\" content=\"en-us\">\r\n<meta name=\"site_name\" content=\"Docs\">\r\n<meta name=\"search.ms_product\" content=\"MSDN\">\r\n<meta name=\"depot_name\" content=\"MSDN.docs-internal\">\r\n<meta name=\"original_ref_skeleton_git_url\" content=\"https://github.com/dotnet/docs-internal/blob/master/api/System.ServiceModel.CommunicationException.yml\">\r\n<meta name=\"APIName\" content=\"System.ServiceModel.CommunicationException\">\r\n<meta name=\"APIName\" content=\"System.ServiceModel.CommunicationException..ctor\">\r\n<meta name=\"APILocation\" content=\"System.ServiceModel.dll\">\r\n<meta name=\"TopicType\" content=\"apiref\">\r\n<meta name=\"APIType\" content=\"Assembly\">\r\n<meta name=\"updated_at\" content=\"2017-03-29 05:54 AM\">\r\n<meta name=\"document_id\" content=\"1448a088-b125-5436-bae4-84794de3fb50\">\r\n<meta name=\"pagetype\" content=\"Reference\">\r\n<meta name=\"description\" content=\"Represents a communication error in either the service or client application.\n\">\r\n<meta name=\"toc_rel\" content=\"_splitted/system.servicemodel/toc2.json\">\r\n<meta name=\"source_url\" content=\"\">\r\n<meta name=\"ms.assetid\" content=\"System.ServiceModel.CommunicationException\">\r\n","rawMetadata":{"author":"dotnet-bot","ms.author":"dotnetcontent","manager":"wpickett","breadcrumb_path":"/dotnet/toc.json","ms.topic":"managed-reference","ms.prod":".net","search.ms_sitename":"Docs","search.ms_docsetname":"docs-internal","version":0,"_op_canonicalUrlPrefix":"https://docs.microsoft.com/en-us/dotnet-internal/","locale":"en-us","site_name":"Docs","search.ms_product":"MSDN","_op_openToPublicContributors":false,"depot_name":"MSDN.docs-internal","_op_gitRefSkeletonCommitHistory":[],"original_ref_skeleton_git_url":"https://github.com/dotnet/docs-internal/blob/master/api/System.ServiceModel.CommunicationException.yml","open_to_public_contributors":false,"api_name":["System.ServiceModel.CommunicationException","System.ServiceModel.CommunicationException..ctor"],"api_location":["System.ServiceModel.dll"],"topic_type":["apiref"],"api_type":["Assembly"],"f1_keywords":["System.ServiceModel.CommunicationException","System::ServiceModel::CommunicationException","System.ServiceModel.CommunicationException.#ctor","System::ServiceModel::CommunicationException::#ctor"],"dev_langs":["csharp"],"updated_at":"2017-03-29 05:54 AM","document_id":"1448a088-b125-5436-bae4-84794de3fb50","content_git_url":"https://github.com/dotnet/docs-internal/blob/master/xml/System.ServiceModel/CommunicationException.xml","layout":"Reference","_op_layout":"Reference","pagetype":"Reference","title":"CommunicationException class | Microsoft Docs","_op_ogTitle":"CommunicationException class","description":"Represents a communication error in either the service or client application.\n","toc_asset_id":"api/_splitted/System.ServiceModel/toc.json","toc_rel":"_splitted/System.ServiceModel/toc.json","source_url":"","ms.assetid":"System.ServiceModel.CommunicationException","canonical_url":"https://docs.microsoft.com/en-us/dotnet/api/system.servicemodel.communicationexception","_op_searchScope":[".NET"],"_op_canonicalUrl":"https://docs.microsoft.com/en-us/dotnet/api/system.servicemodel.communicationexception","fileRelativePath":"api/System.ServiceModel.CommunicationException.html"},"themesRelativePathToOutputRoot":"_themes/"}